src/app/market/buy/buy.component.ts

```diff
@@ -200,6 +236,49 @@ export class BuyComponent implements OnInit {
     }
   }
 
+  static ShippingDetails = class {
+    static firstName: string;
+    static lastName: string;
+    static addressLine1: string;
+    static addressLine2: string;
+    static city: string;
+    static zip: string;
+    static state: string;
+    static country: string;
+  }
+
+  onKeyFirstName(value: string) {
```

Comment:
if we using 'ngModel` then why we need this, ngModel will automatically bind value